HopsanGUI/LibraryHandler.cpp

```diff
@@ -535,6 +536,14 @@ bool LibraryHandler::loadLibrary(SharedComponentLibraryPtrT pLibrary, LibraryTyp
                         sourceElement = sourceElement.nextSiblingElement(QString(XML_LIBRARY_SOURCE));
                     }
 
+                    // Read extra source files
+                    QDomElement extraSourceElement = xmlRoot.firstChildElement(QString(XML_LIBRARY_EXTRA_SOURCE));
```

Comment:
Is extra sources really needed, could just append files to sources

Reply:
It is needed because we don't want to compile the "master source file" for the library when including the components in exported code. An alternative solution could be to have "mainsource" and "source" instead of "source" and "extrasource", but it's just different names.

Reply:
OK I see what you mean. Does this mean that one has to add extra source files to libraries in order for them to be exportable to .exe?

Reply:
No, we generate a new file called `extra-components.cpp`, where we include component .hpp files and register them in HopsanCore.

Reply:
Yes but the reason I ask is because I wrote some of this code maybe a year ago or more and I vagely remember solving this problem somehow without adding extra sources to the xml. But it was so long ago now that I have forgotten what I did. I assume that you are partially using the code I wrote then. I will have a look at the code again and see if I can figure out what I am actually asking about here.

Reply:
OK, lets see if I got this right.  The reason for the "extra sources" field is to enable building of libraries that consists of multiple cpp files. When the library is exported as a bundle (in an FMU or EXE) the "registration code" is auto-generated, and the "main source file" that is listed in sources is not needed, so it is ignored by the export code. The problem with the previous code was that it did not export any of the additional source files at all in this case. So such libraries would not work.  Your addition of "extra-sources" solves this by listing them under a separate tag in the XML?

Reply:
Yes, that is correct. We have probably not noticed it before, since most libraries have only one source file. 

A related problem is how to handle include paths. Right now they are specified by the `<cflags>` tag, usually with a relative path. Extracting paths from this string and converting them to absolute path if needed is for sure possible, but not a very nice solution. Perhaps we should instead add  `<includepath>` tags?

A third problem is how to handle other compile and link flags. We could just append them to the build flags for the executable, but how can we know that they don't conflict with existing flags or libraries?

Reply:
Ok, I have had one such library in the past. But since we only imported it was enough to place all source files in sources. I think that since it is now possible to auto generate the registration code at import/export, maybe in the future there is no need for a "main source".
  The include paths were used for aditional headers that were shipped with the library, so relativ path forks find, but if you depend on headers in your system or other installation path, then that is ofcourse not portable between systems.
 Build flags for different compilers can be enabled I think, but some new code is needed. Right now you can set attributes in the XML element for OS and Architecture (I think). This should also support different compilers.

Reply:
The problem with include paths is that they are relative to the location of the library, not to the build directory of the exported code. If it is specified using build flags we need to extract the code from `-I<path>` and then convert it to an absolute path. If it is instead specified using an <includepath> XML tag we don't need the extraction step, we can just convert it directly. What do you think?

Reply:
A yes, before we built the code directly inside the library directory. Thats why it was not a problem. Yes I think that you are right. We should separate includepaths and also linkpaths (or whatever they should be called)  from the cflags and lflags. In this PR or a different one? If you do it in this one, remember to write both save and load, also for holc so that this data is not lost if library loaded there.

Reply:
I added XML tags for include paths, link paths and link libraries in CAF files. The link libraries maybe don't need their own tag, but it can make it easier for users to understand.